cli/cmd/generate_execute.go

```diff
@@ -370,30 +370,33 @@ func TerraformPlanAndExecute(workingDir string) error {
 		return err
 	}
 
-	// Write plan
-	changes, err := TerraformExecPlan(tf)
-	if err != nil {
-		return err
-	}
+	// If running in nonInteractive mode, dont run a plan or prompt to proceed; just proceed
+	if !cli.nonInteractive {
+		// Write plan
+		changes, err := TerraformExecPlan(tf)
+		if err != nil {
+			return err
+		}
 
-	// Display changes and determine if apply should proceed
-	proceed, err := DisplayTerraformPlanChanges(tf, *changes)
-	if err != nil {
-		return err
-	}
+		// Display changes and determine if apply should proceed
+		proceed, err := DisplayTerraformPlanChanges(tf, *changes)
+		if err != nil {
+			return err
+		}
 
-	// If not proceed; display guidance on how to continue outside of this session
-	if !proceed {
-		provideGuidanceAfterExit(true, true, tf.WorkingDir(), tf.ExecPath())
-		return nil
+		// If not proceed; display guidance on how to continue outside of this session
+		if !proceed {
+			provideGuidanceAfterExit(true, true, tf.WorkingDir(), tf.ExecPath())
```

Comment:
apologies for bringing this up but, can we add unit tests for this function? I don't thin we are testing it and is building some valuable message to the user.

Reply:
👍